File: Unidad 2-Estructuras de Control/Ejercicio6.cs
/*ejercicio 6.-Escribir el programa para un programa en C que reciba tres números
enteros que representan dos catetos y una hipotenusa y que determine si
cumple con la ecuación de pitágoras.
a2 +b2 = c2
Autor: Ronsson Santos
*/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ej6
{
    class Program
    {
        static void Main(string[] args)
        {
            //declaracion de variables
            int hipo, cateto1, cateto2, resultado, lados, hipotenusa;

            //entradas de datos
            Console.WriteLine("Este programa calcula la suma de los cuadrados de 2 catetos \n y los compara con el cuadrado de la hipotenusa\n");
            Console.WriteLine("para testear nuestro programa en catetos puedes usar los valores 3,4\n");
            Console.WriteLine("y para la hipotenusa puedes usar el valor 5\n");
            Console.WriteLine("Digite los 2 catetos separados por un espacio: ");
            cateto1= Convert.ToInt32(Console.ReadLine());
            cateto2 = Convert.ToInt32(Console.ReadLine());
            Console.WriteLine("Ahora digite la hipotenusa : ");
            hipo = Convert.ToInt32(Console.ReadLine());

            //proceso de informacion
            lados = ((cateto1*cateto1) + (cateto2*cateto2));
            hipotenusa = hipo * hipo;
            Console.WriteLine("-----------------------------------------------------\n");

            //comparar a2=b2+c2
            if (lados == hipotenusa)
            {
                Console.WriteLine("Los datos introducidos corresponden a un triangulo Rectangulo");
                Console.ReadKey();
            }
            else
            {
                Console.WriteLine("Los valores introducidos NO corresponden a un triangulo rectangulo");
                Console.ReadKey();
            }


        }
    }
}
/*
QA
Autor QA:Eyder Antonio Concha Moreno
Entradas: (3, 4, 5), (9, 10, 5)
salidas: "Los datos corresponden a un triC!ngulo RectC!ngulo" Correcto, "Los valores introducidos NO corresponden a un triangulo rectangulo" Correcto
Proceso: Es correcto, sin embargo debería ser capaz de funcionar con valores indeterminados al inicio
*/

File: Unidad 3-Funciones/ejercicio6.js
/*ejercicio 6.-Escribir con modulos un programa en C que reciba tres números
enteros que representan dos catetos y una hipotenusa y que determine si
cumple con la ecuación de pitágoras.
a2 +b2 = c2
Autor: Ronsson Ramiro may santos
fecha. 17/febrero/2019
*/
let valCat1,valCat2,hipotenusa,resultado;
valCat1=getValueCat1();
valCat2=getvalueCat2();
hipotenusa=getHipotenusa();
resultado=catetos(valCat1,valCat2);
alert("LA RAIZ CUADRADA DE LOS 2 CATETOS AL CUADRADO ES : "+ resultado);
comparacion( resultado,hipotenusa);

function getValueCat1(){// funcion numero 1 --obtiene el valor del cateto1
    valCat1=Number(prompt("INTRODUCE EL VALOR DEL CATETO 1:"));
    return valCat1;
}

function getvalueCat2(){//  funcion numero 2--obtiene el valor del cateto1
    valCat2=Number(prompt("INTRODUCE EL VALOR DEL CATETO 2:"));
    return valCat2;
}

function catetos( cateto1, cateto2){//  funcion numero 3 --obtiene la raiz del cuadrado de los catetos
raizCatetosCuadrados=Math.sqrt(Math.pow(cateto1,2)+Math.pow(cateto2,2));/*sqrt(pow(cateto1,2)+pow(cateto2,2));*/
return raizCatetosCuadrados;
}
function getHipotenusa(){ //  funcion numero 4 --obtiene el valor de la hipotenusa
    hipotenusa = Number(prompt("INTRODUCE EL VALOR DE LA HIPOTENUSA:"));
    return hipotenusa;
}
function comparacion( resultado, hipotenusa){ // producto del proceso
    //resultado=Math.sqrt(Math.pow(valCat1,2)+Math.pow(valCat2,2));
    if(resultado==hipotenusa){
        alert("LOS VALORES INTRODUCIDOS CORRESPONDEN A UN TRIANGULO RECTANGULO");
        return ;
    }else{
        alert("Los valores introducidos NO corresponden a un triangulo rectangulo");
        return;
    }
    return ;
}
/*
    QA Reviso: Carlos Chan 
    
    getValueCat1():
    Entradas: 5, 10, 3
    Salidas: 5, 10, 3
    Ok
    
    getValueCat2():
    Entradas: 10, 3, 4
    Salidas: 10, 3, 4
    Ok
    
    getHipotenusa():
    Entradas: 12, 11, 8
    Salidas: 12, 11, 8
    Ok
    
    catetos():
    Entradas: 10, 4
    Salidas: 10.77
    Ok
    
    comparacion():
    Entradas: (10.77, 15), (5, 5)
    Salidas: Ok, Ok
    
    Integracion:
    Entradas: (6, 8, 10), (12, 16, 20), (1, 3, 7)
    Salidas: Ok, Ok, Ok
*/
